### ISSUE TITLE
ModuleNotFoundError: No module named ‘NaiveDE’

### ISSUE THREAD

**Issue by kinnaryshah** [sales-lab/spatialDE#26]
Hello spatialDE package developer,

I wanted to bring to your attention an issue that I am running into when trying to use this package. I was not able to run the spatialDE function on my SpatialExperiment object and wanted to test if the example would run. After I run this code to use the example:

```r
library(spatialDE)
example(spatialDE)
```

I run into this error:

```
Error in py_module_import(module, convert = convert) :
  ModuleNotFoundError: No module named ‘NaiveDE’
Run `reticulate::py_last_error()` for details.

> `reticulate::py_last_error()`
── Python Exception Message ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Traceback (most recent call last):
  File “/jhpce/shared/community/core/conda_R/4.3.x/R/lib64/R/site-library/reticulate/python/rpytools/loader.py”, line 119, in _find_and_load_hook
   return _run_hook(name, _hook)
  File “/jhpce/shared/community/core/conda_R/4.3.x/R/lib64/R/site-library/reticulate/python/rpytools/loader.py”, line 93, in _run_hook
   module = hook()
  File “/jhpce/shared/community/core/conda_R/4.3.x/R/lib64/R/site-library/reticulate/python/rpytools/loader.py”, line 117, in _hook
   return _find_and_load(name, import_)
ModuleNotFoundError: No module named ‘NaiveDE’
── R Traceback ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   ▆
1. ├─utils::example(spatialDE)
2. │ └─base::source(...)
3. │ ├─base::withVisible(eval(ei, envir))
4. │ └─base::eval(ei, envir)
5. │  └─base::eval(ei, envir)
6. ├─spatialDE::spatialDE(spe) at tmp/RtmpakGLLB/Rex340627e6f89a9:13:1
7. └─spatialDE::spatialDE(spe)
8. └─spatialDE (local) .local(x, ...)
9.  └─spatialDE:::.spatialDE(...)
10.   └─spatialDE:::.naiveDE_stabilize(proc, x)
11.    └─basilisk::basiliskRun(...)
12.     └─spatialDE (local) fun(..., store = proc)
13.      └─reticulate::import(“NaiveDE”)
14.       └─reticulate:::py_module_import(module, convert = convert)
```

I’ve included the session info below.

```
`py_config()`
python:    /users/kshah/.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env/bin/python
libpython:  /users/kshah/.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env/lib/libpython3.9.so
pythonhome:  /users/kshah/.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env:/users/kshah/.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env
version:    3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:22:55) [GCC 10.3.0]
numpy:     /users/kshah/.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env/lib/python3.9/site-packages/numpy
numpy_version: 1.26.3
NOTE: Python version was forced by use_python() function

`sessionInfo()`
R version 4.3.2 Patched (2023-11-13 r85524)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: Rocky Linux 9.2 (Blue Onyx)
Matrix products: default
BLAS: /jhpce/shared/community/core/conda_R/4.3.x/R/lib64/R/lib/libRblas.so
LAPACK: /jhpce/shared/community/core/conda_R/4.3.x/R/lib64/R/lib/libRlapack.so; LAPACK version 3.11.0locale:
[1] LC_CTYPE=en_US.UTF-8   LC_NUMERIC=C
[3] LC_TIME=en_US.UTF-8    LC_COLLATE=en_US.UTF-8
[5] LC_MONETARY=en_US.UTF-8  LC_MESSAGES=en_US.UTF-8
[7] LC_PAPER=en_US.UTF-8   LC_NAME=C
[9] LC_ADDRESS=C       LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C
 time zone: US/Eastern
tzcode source: system (glibc)
attached base packages:
[1] stats4  stats  graphics grDevices datasets utils  methods
[8] base
other attached packages:
[1] reticulate_1.34.0     SpatialExperiment_1.12.0
[3] spatialDE_1.8.0      scran_1.30.0
[5] scuttle_1.12.0       SingleCellExperiment_1.24.0
[7] SummarizedExperiment_1.32.0 Biobase_2.62.0
[9] GenomicRanges_1.54.1    GenomeInfoDb_1.38.1
[11] IRanges_2.36.0       S4Vectors_0.40.1
[13] BiocGenerics_0.48.1    MatrixGenerics_1.14.0
[15] matrixStats_1.1.0
loaded via a namespace (and not attached):
[1] dir.expiry_1.10.0    gtable_0.3.4
[3] rjson_0.2.21       ggplot2_3.4.4
[5] lattice_0.22-5      generics_0.1.3
[7] vctrs_0.6.4       tools_4.3.2
[9] bitops_1.0-7       parallel_4.3.2
[11] tibble_3.2.1       fansi_1.0.5
[13] cluster_2.1.4      pkgconfig_2.0.3
[15] BiocNeighbors_1.20.0   Matrix_1.6-3
[17] checkmate_2.3.0     sparseMatrixStats_1.14.0
[19] dqrng_0.3.1       lifecycle_1.0.4
[21] GenomeInfoDbData_1.2.11 compiler_4.3.2
[23] statmod_1.5.0      munsell_0.5.0
[25] bluster_1.12.0      codetools_0.2-19
[27] RCurl_1.98-1.13     pillar_1.9.0
[29] crayon_1.5.2       BiocParallel_1.36.0
[31] DelayedArray_0.28.0   limma_3.58.1
[33] magick_2.8.1       abind_1.4-5
[35] basilisk_1.14.0     tidyselect_1.2.0
[37] rsvd_1.0.5        locfit_1.5-9.8
[39] metapod_1.10.0      BiocSingular_1.18.0
[41] dplyr_1.1.3       grid_4.3.2
[43] colorspace_2.1-0     cli_3.6.1
[45] SparseArray_1.2.2    magrittr_2.0.3
[47] S4Arrays_1.2.0      utf8_1.2.4
[49] withr_2.5.2       edgeR_4.0.1
[51] backports_1.4.1     filelock_1.0.2
[53] DelayedMatrixStats_1.24.0 scales_1.2.1
[55] XVector_0.42.0      igraph_1.5.1
[57] gridExtra_2.3      png_0.1-8
[59] ScaledMatrix_1.10.0   beachmat_2.18.0
[61] basilisk.utils_1.14.0  irlba_2.3.5.1
[63] rlang_1.1.2       Rcpp_1.0.11
[65] glue_1.6.2        BiocManager_1.30.22
[67] jsonlite_1.8.7      R6_2.5.1
[69] zlibbioc_1.48.0
```

While reinstalling the package and running the function the first time, it puts the output of installing the Python packages in a cache. Below is from @ kasperdanielhansen session: 

```
> `example(spatialDE)`

sptlDE> ## Mock up a SpatialExperiment object wit 400 cells and 3 genes
sptlDE> set.seed(42)

sptlDE> spe <- mockSVG(size = 20, tot_genes = 3, de_genes = 1, return_SPE = TRUE)
sptlDE> ## Run spatialDE
sptlDE> de_results <- spatialDE(spe)
trying URL 'https://repo.anaconda.com/miniconda/Miniconda3-py39_4.12.0-Linux-x86_64.sh'
Content type 'application/x-sh' length 76607678 bytes (73.1 MB)
==================================================
downloaded 73.1 MB

PREFIX=/users/khansen/.cache/R/basilisk/1.14.0/0
Unpacking payload ...
Collecting package metadata (current_repodata.json): done
Solving environment: done
```

in between here is conda installations

```
....



libblas-3.9.0        | 14 KB     | ############################################################################################################################################################################################################################## | 100%
pytz-2024.1          | 184 KB    | ############################################################################################################################################################################################################################## | 100%
scipy-1.12.0         | 15.6 MB   | ############################################################################################################################################################################################################################## | 100%
libgfortran-ng-13.2. | 23 KB     | ############################################################################################################################################################################################################################## | 100%
libopenblas-0.3.26   | 5.3 MB    | ############################################################################################################################################################################################################################## | 100%
six-1.16.0           | 14 KB     | ############################################################################################################################################################################################################################## | 100%
python-dateutil-2.8. | 240 KB    | ############################################################################################################################################################################################################################## | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
Collecting SpatialDE==1.1.3
  Downloading SpatialDE-1.1.3-py3-none-any.whl (15 kB)
Collecting NaiveDE==1.2.0
  Downloading NaiveDE-1.2.0.tar.gz (4.8 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy in ./.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env/lib/python3.9/site-packages (from SpatialDE==1.1.3) (1.23.5)
Requirement already satisfied: scipy>=1.0 in ./.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env/lib/python3.9/site-packages (from SpatialDE==1.1.3) (1.12.0)
Requirement already satisfied: pandas>=0.23 in ./.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env/lib/python3.9/site-packages (from SpatialDE==1.1.3) (1.5.2)
Collecting tqdm (from SpatialDE==1.1.3)
  Downloading tqdm-4.66.1-py3-none-any.whl.metadata (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.6/57.6 kB 3.4 MB/s eta 0:00:00
Collecting Click (from SpatialDE==1.1.3)
  Downloading click-8.1.7-py3-none-any.whl.metadata (3.0 kB)
Requirement already satisfied: python-dateutil>=2.8.1 in ./.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env/lib/python3.9/site-packages (from pandas>=0.23->SpatialDE==1.1.3) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in ./.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env/lib/python3.9/site-packages (from pandas>=0.23->SpatialDE==1.1.3) (2024.1)
Requirement already satisfied: six>=1.5 in ./.cache/R/basilisk/1.14.0/spatialDE/1.8.0/env/lib/python3.9/site-packages (from python-dateutil>=2.8.1->pandas>=0.23->SpatialDE==1.1.3) (1.16.0)
Downloading click-8.1.7-py3-none-any.whl (97 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.9/97.9 kB 4.3 MB/s eta 0:00:00
Downloading tqdm-4.66.1-py3-none-any.whl (78 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.3/78.3 kB 2.8 MB/s eta 0:00:00
Building wheels for collected packages: NaiveDE
  Building wheel for NaiveDE (setup.py) ... done
  Created wheel for NaiveDE: filename=NaiveDE-1.2.0-py3-none-any.whl size=5688 sha256=c3630d704dfd9a4e046fb51b3427acc1ecc280302548bd09c2f387e0efa2d177
  Stored in directory: /users/khansen/.cache/pip/wheels/82/8e/4e/dd6448962915d382128e962215e43abb69f8f1ab24adb98825
Successfully built NaiveDE
Installing collected packages: tqdm, Click, NaiveDE, SpatialDE
Successfully installed Click-8.1.7 NaiveDE-1.2.0 SpatialDE-1.1.3 tqdm-4.66.1
Models: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 35.99it/s]
Error in py_call_impl(callable, call_args$unnamed, call_args$named) :
  AttributeError: Module 'scipy' has no attribute 'argsort'
Run `reticulate::py_last_error()` for details.

Enter a frame number, or 0 to exit

 1: example(spatialDE)
 2: source(tf, local, echo = echo, prompt.echo = paste0(prompt.prefix, getOptio
 3: withVisible(eval(ei, envir))
 4: eval(ei, envir)
 5: eval(ei, envir)
 6: Rex1e229734e0d239#13: spatialDE(spe)
 7: spatialDE(spe)
 8: .local(x, ...)
 9: .spatialDE(x = assay(x, assay_type), coordinates = coordinates, verbose = v
10: .spatialDE_run(proc, regressed, coords)
11: basiliskRun(proc, function(x, coordinates, store) {
    spatialDE <- store$s
12: fun(..., store = proc)
13: spatialDE$run(X, res_py)
14: py_call_impl(callable, call_args$unnamed, call_args$named)

Selection:
```

**Comment by gbrsales** [comment]
Hi @kinnaryshah,

I recently released a new version of the package (1.8.1) pinning a specific version of `scipy`. This should fix the error you encountered.

Can you try upgrading?

**Comment by kinnaryshah** [comment]
thank you!
